cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -448,6 +445,92 @@ void build_host(raft::resources const& handle,
                 raft::host_matrix_view<const uint8_t, int64_t, raft::row_major> dataset,
                 cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx);
 
+/**
+ * @defgroup cagra_cpp_index_add_nodes CAGRA graph node addition functions
+ * @{
+ */
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::device_mdspan<const float, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<float, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::host_mdspan<const float, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<float, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::device_mdspan<const int8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<int8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::host_mdspan<const int8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<int8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::device_mdspan<const uint8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+void add_graph_nodes(
+  raft::resources const& handle,
+  raft::host_mdspan<const uint8_t, raft::matrix_extent<int64_t>, raft::layout_stride>
+    updated_dataset_view,
+  const cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx,
+  raft::host_matrix_view<uint32_t, std::int64_t> updated_graph_view,
+  uint32_t batch_size = 0);
+
+/**
+ * @defgroup cagra_cpp_index_extend CAGRA extend functions
+ * @{
+ */
+void extend(raft::resources const& handle,
+            raft::device_matrix_view<const float, int64_t, raft::row_major> additional_dataset,
+            cuvs::neighbors::cagra::index<float, uint32_t>& idx,
+            uint32_t batch_size = 0);
+
+void extend(raft::resources const& handle,
+            raft::host_matrix_view<const float, int64_t, raft::row_major> additional_dataset,
+            cuvs::neighbors::cagra::index<float, uint32_t>& idx,
+            uint32_t batch_size = 0);
```

Comment:
I'm a little torn about this batch size parameter- is this something we can infer? Without docs, it's hard for me to know what it means or how to set it so maybe that will help. In the absence of this knowledge, though, we are generally trying to keep the "build", "extend", and "search" APIs as consistent as possible across all the algorithms so anything we can do to minimize differences is helpful.

Reply:
I replaced the `chunk_size` param with `extend_params`.